include/swift/AST/Types.h

```diff
@@ -2297,6 +2297,16 @@ class AnyFunctionType : public TypeBase {
     return getExtInfo().throws();
   }
 
+  unsigned getCurryLevel() const {
```

Comment:
I think you should use the decl's getNumParameterLists() instead. If a function is declared as returning a function, we do not want to flatten it's type (eg func foo() -> () -> ())